pkg/network/usm/postgres_monitor_test.go

```diff
@@ -0,0 +1,313 @@
+	cfg.EnablePostgresMonitoring = true
+	cfg.MaxTrackedConnections = 1000
+	return cfg
+}
+
+func prepareTestDB(t *testing.T, ctx testContext) {
+	postgres.ConnectAndGetDB(t, ctx.serverAddress, ctx.extras)
+	postgres.RunCreateQuery(t, ctx.extras)
+	values := make([]string, 5)
+	for i := 0; i < 20; i++ {
+		for j := 0; j < len(values); j++ {
+			values[j] = fmt.Sprintf("value-%d", i*len(values)+j+1)
+		}
+		postgres.RunMultiInsertQuery(t, ctx.extras, values...)
+	}
+}
+
+func TestPostgresMonitoring(t *testing.T) {
+	skipTestIfKernelNotSupported(t)
+
+	ebpftest.TestBuildModes(t, []ebpftest.BuildMode{ebpftest.Prebuilt, ebpftest.RuntimeCompiled, ebpftest.CORE}, "", func(t *testing.T) {
+		suite.Run(t, new(postgresProtocolParsingSuite))
+	})
+}
+
+func (s *postgresProtocolParsingSuite) TestLoadPostgresBinary() {
+	t := s.T()
+	for name, debug := range map[string]bool{"enabled": true, "disabled": false} {
+		t.Run(name, func(t *testing.T) {
+			cfg := getPostgresDefaultTestConfiguration()
+			cfg.BPFDebug = debug
+			setupUSMTLSMonitor(t, cfg)
+		})
+	}
+}
+
+func validatePostgres(t *testing.T, monitor *Monitor, expectedStats map[string]map[postgres.Operation]int) {
+	found := make(map[string]map[postgres.Operation]int)
+	require.Eventually(t, func() bool {
+		postgresProtocolStats, exists := monitor.GetProtocolStats()[protocols.Postgres]
+		if !exists {
+			return false
+		}
+		// We might not have postgres stats, and it might be the expected case (to capture 0).
+		currentStats := postgresProtocolStats.(map[postgres.Key]*postgres.RequestStat)
+		for key, stats := range currentStats {
+			if _, ok := found[key.TableName]; !ok {
+				found[key.TableName] = make(map[postgres.Operation]int)
+			}
+			found[key.TableName][key.Operation] += stats.Count
+		}
+		return reflect.DeepEqual(expectedStats, found)
+	}, time.Second*5, time.Millisecond*100, "Expected to find a %v stats, instead captured %v", &expectedStats, &found)
+}
+
+func (s *postgresProtocolParsingSuite) TestDecoding() {
+	t := s.T()
+
+	//clientHost := "localhost"
+	targetHost := "127.0.0.1"
+	serverHost := "127.0.0.1"
+
+	serverAddress := net.JoinHostPort(serverHost, postgresPort)
+	targetAddress := net.JoinHostPort(targetHost, postgresPort)
+	require.NoError(t, postgres.RunServer(t, serverHost, postgresPort))
+
+	//defaultDialer := &net.Dialer{
+	//	LocalAddr: &net.TCPAddr{
+	//		IP: net.ParseIP(clientHost),
+	//	},
+	//}
+
+	tests := []postgresParsingTestAttributes{
+		{
+			name: "create table",
+			preMonitorSetup: func(t *testing.T, ctx testContext) {
+				postgres.ConnectAndGetDB(t, ctx.serverAddress, ctx.extras)
+			},
+			postMonitorSetup: func(t *testing.T, ctx testContext) {
+				postgres.RunCreateQuery(t, ctx.extras)
+			},
+			validation: func(t *testing.T, ctx testContext, monitor *Monitor) {
+				validatePostgres(t, monitor, map[string]map[postgres.Operation]int{
+					"dummy": {
+						postgres.CreateTableOP: 2,
+					},
+				})
+			},
+		},
+		{
+			name: "insert",
```

Comment:
Perhaps we could give the test a more descriptive name that explains its purpose?